client/settings/fraud-protection/interfaces.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { Dispatch, SetStateAction } from 'react';
+
+export interface FraudPreventionSetting {
+	block: boolean;
+	enabled: boolean;
+	min_items?: string | number | null;
+	max_items?: string | number | null;
+	min_amount?: string | number | null;
+	max_amount?: string | number | null;
+}
+
+export type ProtectionSettingsUI = Record< string, FraudPreventionSetting >;
+
+export interface FraudPreventionSettingsContextType {
+	protectionSettingsUI: ProtectionSettingsUI;
+	setProtectionSettingsUI: ( settings: ProtectionSettingsUI ) => void;
+	protectionSettingsChanged: boolean;
+	setProtectionSettingsChanged: Dispatch< SetStateAction< boolean > >;
+}
+
+export interface FraudProtectionSettingsSingleCheck {
+	key: string;
+	value: any;
+	operator: string;
+}
+
+export interface FraudProtectionSettingsMultipleChecks {
+	operator: string;
+	checks: FraudProtectionSettingsSingleCheck[];
+}
+
+export type FraudProtectionSettingsCheck =
+	| FraudProtectionSettingsSingleCheck
+	| FraudProtectionSettingsMultipleChecks
+	| null;
+
+export interface AdvancedFraudProtectionSettings {
+	key: string;
+	outcome: string;
+	check: FraudProtectionSettingsCheck;
+}
+
+export type UseCurrentProtectionLevel = [ string, ( level: string ) => void ];
```

Comment:
Is this naming standard in our code base?

Reply:
I'm not sure I understand. Could you expand on this? Since it is a type, I thought that it would need to be in `UpperCamelCase`.

Reply:
I'm not sure I understand. Could you expand on this? Since it is a type, I thought that it would need to be in `UpperCamelCase`.

Reply:
Okay, I understand what you mean now! I like `UseCurrentProtectionLevelHook`, that way it maintains the full name of the hook but includes that suffix saying what it is (a Redux hook) and makes the type a noun. 

WDYT? Thumbs up for `UseCurrentProtectionLevelHook`?

Reply:
I think if we get rid of the `Use` prefixes, and append `Hook`, they may look better.

Reply:
Got it. Fixed in [1b56724](https://github.com/Automattic/woocommerce-payments/pull/6032/commits/1b56724105f1379f1c84461174a4ec0a8ae72a96).

Reply:
I was talking about *all* variables that have the `UseXXX` pattern, you only changed one of them :)

Reply:
🤦 Thanks for pointing that out! I've addressed it in [394a1a3](https://github.com/Automattic/woocommerce-payments/pull/6032/commits/394a1a3cfa4bdb873a5e8ed28309256e64053ca4).